tests/sqlparser_postgres.rs

```diff
@@ -4998,3 +4998,51 @@ fn arrow_cast_precedence() {
         }
     )
 }
+
```

Comment:
could we add the tests in the generic style [like so](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/tests/sqlparser_common.rs#L2765) so that it covers all and future dialects that support this syntax?

Reply:
done.